client/src/components/AlertDialog.jsx

```diff
@@ -47,6 +47,9 @@ export default function AlertDialog({
   useEffect(() => {
         applicationsAPI.getStudentApplications().then((response) => {
           setIsAppliedProposal(response.filter((o) => o.status !== 'rejected').length > 0);
+
+          // Disables the "Apply" button for proposals that have already been applied and rejected.
+          setIsAppliedProposal(response.filter((o) => o.title === item.title && o.status === 'rejected').length > 0);
```

Comment:
But in this way the line 52 overrides the line 49 and it means that the the button is disabled only when the student does an apply for the proposal and the professor reject it

Reply:
Thanks, I fixed this and now both conditions are AND.